pkg/querier/querier.go

```diff
@@ -353,36 +352,48 @@ func (q *SingleTenantQuerier) Label(ctx context.Context, req *logproto.LabelRequ
 	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(q.cfg.QueryTimeout))
 	defer cancel()
 
+	g, ctx := errgroup.WithContext(ctx)
+
 	ingesterQueryInterval, storeQueryInterval := q.buildQueryIntervals(*req.Start, *req.End)
 
 	var ingesterValues [][]string
 	if !q.cfg.QueryStoreOnly && ingesterQueryInterval != nil {
-		timeFramedReq := *req
-		timeFramedReq.Start = &ingesterQueryInterval.start
-		timeFramedReq.End = &ingesterQueryInterval.end
+		g.Go(func() error {
+			var err error
+			timeFramedReq := *req
+			timeFramedReq.Start = &ingesterQueryInterval.start
+			timeFramedReq.End = &ingesterQueryInterval.end
 
-		ingesterValues, err = q.ingesterQuerier.Label(ctx, &timeFramedReq)
-		if err != nil {
-			return nil, err
-		}
+			ingesterValues, err = q.ingesterQuerier.Label(ctx, &timeFramedReq)
+			return err
+		})
 	}
 
 	var storeValues []string
 	if !q.cfg.QueryIngesterOnly && storeQueryInterval != nil {
-		from := model.TimeFromUnixNano(storeQueryInterval.start.UnixNano())
-		through := model.TimeFromUnixNano(storeQueryInterval.end.UnixNano())
-
-		if req.Values {
-			storeValues, err = q.store.LabelValuesForMetricName(ctx, userID, from, through, "logs", req.Name)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			storeValues, err = q.store.LabelNamesForMetricName(ctx, userID, from, through, "logs")
-			if err != nil {
-				return nil, err
+		g.Go(func() error {
+			var err error
+			from := model.TimeFromUnixNano(storeQueryInterval.start.UnixNano())
+			through := model.TimeFromUnixNano(storeQueryInterval.end.UnixNano())
+
+			if req.Values {
+				storeValues, err = q.store.LabelValuesForMetricName(ctx, userID, from, through, "logs", req.Name)
+				if err != nil {
+					return err
+				}
+			} else {
+				storeValues, err = q.store.LabelNamesForMetricName(ctx, userID, from, through, "logs")
+				if err != nil {
+					return err
+				}
 			}
-		}
+
+			return nil
```

Comment:
could we just return `err` here and remove the 2 nested `if err != nil` checks? Since they're in an if/else only one branch should execute, yeah?

Reply:
[965fec7](https://github.com/grafana/loki/pull/5997/commits/965fec75a83f7882eccb88b768abbb4f8ebd5674)